src/main/java/book/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package book.store.service.impl;
+
+import book.store.dto.CartItemRequestDto;
+import book.store.dto.ShoppingCartResponseDto;
+import book.store.mapper.CartItemMapper;
+import book.store.mapper.ShoppingCartMapper;
+import book.store.model.CartItem;
+import book.store.model.ShoppingCart;
+import book.store.model.User;
+import book.store.repository.BookRepository;
+import book.store.repository.CartItemRepository;
+import book.store.repository.ShoppingCartRepository;
+import book.store.service.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public void createCartItem(CartItemRequestDto cartItemRequestDto,
+                               Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        CartItem cartItem = toCartItem(cartItemRequestDto,user);
+        cartItemRepository.save(cartItem);
+    }
+
+    @Override
+    public ShoppingCartResponseDto getShoppingCart(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findShoppingCartByUserId(user.getId())
+                .orElseThrow();
+        return shoppingCartMapper.toShoppingCartResponseDto(shoppingCart);
+    }
+
+    @Override
+    public void deleteCartItem(Long id,Authentication authentication) {
+        if (checkCartItemOwnerMatching(id,authentication)) {
+            cartItemRepository.deleteById(id);
+        } else {
+            throw new RuntimeException("User is not owned by this shopping cart.");
+        }
```

Comment:
```suggestion
        if (!checkCartItemOwnerMatching(id,authentication)) {
              throw new RuntimeException("User is not owned by this shopping cart.");
         }
         cartItemRepository.deleteById(id);            
      }
```

Reply:
in cases like this, invert condition in if, throw/return early, and you won't need else clause, this code is easier to read